packages/shared/lib/ledger.ts

```diff
@@ -38,16 +43,38 @@ export function calculateLedgerDeviceState(status: LedgerStatus): LedgerDeviceSt
     if (locked) {
         return LedgerDeviceState.Locked
     } else {
-        if (app?.name === AppName.IOTA) {
-            return LedgerDeviceState.Connected
-        } else if (app?.name === AppName.IOTALegacy) {
-            return LedgerDeviceState.LegacyConnected
-        } else {
-            return connected ? LedgerDeviceState.AppNotOpen : LedgerDeviceState.NotDetected
+        switch(app?.name) {
+            default:
+                if(connected) {
+                    /**
+                     * NOTE: "BOLOS" is the name of the Ledger operating system and is
+                     * sometimes registered as an app.
+                     */
+                    return (app?.name && app?.name !== 'BOLOS') ? LedgerDeviceState.OtherConnected : LedgerDeviceState.AppNotOpen
```

Comment:
I would suggest defining `BOLOS` as a constant and then using it here. That way it could be reused.